department-of-veterans-affairs/abd-vro
# MCP 2324:  Set partial date from MAS

## What was the problem?


Associated tickets or Slack threads:

- [MCP-2324](https://amida.atlassian.net/browse/MCP-2324)
Partial date logic was complicated and did not use MAS "partialDate" (discovered in most recent sample data).

## How does this fix it?[^1]

Takes "partialDate" from MAS and use it to populate the PDF, when there is no parse-able date. Safely handles dates from MAS that are not in the expected format (%Y-%m-%d).

## How to test this PR
- Step 1 Run Pyhton tests,
- Build app and  try the PDF with partial date
- Step 2[^secrel]
- Copy and paste test data from resources/test-data/pdf-generator-mas.json and set "pdfTemplate":"v2" in the request body
- See that one of the medications says "No confident date ..."

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
